### DISCUSSION TITLE
Incorporation of cropping video frames via x,y, width, and height values

### DISCUSSION THREAD

**Discussion by daskro** [vzhd1701/gridplayer#179] [Ideas]
The idea would be to allow for user to allow the cropping of individual video frames by controlling the values of x,y, width and height.  This enables the user to zoom in on specific areas of the video as well as allow for custom aspect ratios.
Control could be done a few ways:
- Via the context menu similar to the already incorporated zoom in the context menu
- Via hotkeys, such as "u" & "i" for X +1 & X -1, "o" & "p" for Y +1 & Y -1, "shift + u" & "shift + i" for width +1 & width -1, "shift + o" & "shift + p" for height +1 & height -1, 
- VLC has an "interactive zoom" feature creates an interactive overlay to zoom in on a specific area of the video however this feature always maintains the video's aspect ratio. Overlay could be toggled via hotkey similar to the play and timeline overlays.

Other features could include:

- Save and load custom x,y,width, and height cropping values for each video in a playlist.
- Toggle ability to lock video's original aspect ratio similar to VLC's interactive zoom.

**Comment by vzhd1701** [comment]
I have implemented cropping in the new version, please check it out and tell me how it works for you.

My implementation is slightly different. I am using left-top-right-bottom cropping strategy, it is more convenient since I don't need to store video size. It also works better when using VLC scaling in combination with this feature. Your original idea (WxH+X+Y) would require replacing the VLC builtin scaling mechanism with doing the zoom using just the crop function, otherwise it would not output to the right portion of the video set in the XY offset.

Also the aspect ratio will automatically switch to "Aspect None" when using crop function. This is similar to how VLC operates since aspect cropping is done via the crop function.

Since Shift+<key> hotkeys are dedicated to bulk video commands, I used Alt+uiop for right and bottom crop, I hope that's OK. I will implement hotkey editor later.

Unfortunately I don't have dynamic access to VLC interactive zoom video filter via libVLC, so adding overlay would be too messy with current implementation.
